### DISCUSSION TITLE
Prefer highest % match first?

### DISCUSSION THREAD

**Discussion by rieje** [junegunn/fzf#3962] [Q&A] [answered]
I have the filter as: `home jacob dev`. It prefers the entry:

> /home/jacob/misc/file-archive/e-wd1000/resources/programming/miscellaneous/Python Desktop Application Development

over 

> /home/jacob/dev/

Is it possible to prefer the latter and/or is there any trick to e.g. prefer certain directories for this (except outright excluding the directory)?

Thanks!

**Comment by junegunn** [comment, marked as answer]
Try entering the query without spaces. i.e. `homejacobdev` or even `hojade`.

https://junegunn.github.io/fzf/search-syntax/#how-to-get-better-results

**Reply by rieje** [reply]
Thanks. Do you know how I can use `cmd` function for `reload` in this [shell script](https://0x0.st/XWB6.txt)?

**Reply by junegunn** [reply]
Export the function. i.e. `export -f cmd`

**Reply by rieje** [reply]
Same as without, it works in the command substitution but  not inside as `reload(cmd ...` with the error:

> [Command failed: cmd /home ...]

when I  toggle with Ctrl-r.

**Reply by LangLangBart** [reply]
> it works in the command substitution but not inside as reload(cmd ... with the error:

>[!NOTE] 
> fzf starts child processes using `$SHELL -c` if `$SHELL` is set, otherwise sh -c.

Could you try prefixing your `fzf` command with the `SHELL` environment variable assigned to `bash`? 
Alternatively, using `--with-shell="$(which bash) -c"` should work too, added in version `0.51.0`[^1].

```diff
--- a/script
+++ b/script
  fd -t d . "$@" --hidden --no-ignore -E '.git' --follow
 }

-selection=$(cmd "$HOME" /media | fzf --preview 'tree -C {}' \
+export -f cmd
+selection=$(cmd "$HOME" /media | SHELL="$(which bash)" fzf --preview 'tree -C {}' \
  --header 'cd to selection (Ctrl-r to toggle filtering for non-media dirs only)' \
  --prompt 'Dirs > ' \

```

[^1]: [fzf/CHANGELOG.md at master · junegunn/fzf · GitHub](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0510)

**Reply by rieje** [reply]
That solved it, thank you!
